File: 22.py
#!/usr/bin/env python

"""This script solves the Project Euler problem "Names scores". The problem is:
What is the total of all the name scores in the file?
"""

import csv
try:
    from urllib.request import urlopen
except ImportError:
    from urllib import urlopen
try:
    from io import StringIO
except ImportError:
    from StringIO import StringIO


def main():
    """Names scores"""

    # Constants
    NAMES_URL = 'https://projecteuler.net/project/resources/p022_names.txt'

    response = urlopen(NAMES_URL)
    names = response.read()
    names = [row for row in csv.reader(StringIO(names.decode('utf-8')))][0]

    total = 0

    for i, name in enumerate(sorted(names), start=1):
        value = sum([ord(letter) - 64 for letter in name])
        total += i * value

    print(total)

if __name__ == '__main__':
    main()

File: 22.js
#!/usr/bin/env node

// Names scores

var NAMES_URL = 'https://projecteuler.net/project/resources/p022_names.txt';

var request = require('request');
request(NAMES_URL, function(error, response, body) {
    if (!error && response.statusCode == 200) {
        score_names(body.replace(/"/g, "").split(",").sort());
    }
});

function score_names(names) {
    var total = 0;

    for (var i = 0; i < names.length; i++) {
        var value = 0;
        for (var j = 0; j < names[i].length; j++) {
            value += names[i].charCodeAt(j) - 64;
        }
        total += (i + 1) * value;
    }

    console.log(total);
}
